src/QueryBuilder/QueryBuilder/helpers/valueBuilder.js

```diff
@@ -27,17 +27,21 @@ export const getFormattedUUID = (value, isInRelatedOperator) => {
     : getQuotedStr(value);
 };
 
-const formatDateToPreview = (dateString) => {
-  const date = moment(dateString);
-
-  if (date.isValid()) {
-    return date.format(ISO_FORMAT);
+const formatDateToPreview = (dateString, intl) => {
+  const formatedDate = moment(dateString); 
+  if (formatedDate.isValid()) {
+    return intl.formatDate(formatedDate.toDate(),
+     {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit'
+    });
   }
-
   return dateString;
 };
 
 export const valueBuilder = ({ value, field, operator, fieldOptions }) => {
+  const intl = useIntl();
```

Comment:
You cannot bring `intl` in here; it must be passed in by the caller. It can only be introduced in a React component or other hook, so you'll have to pass it in here via `query.js@getQueryStr`, which has to get it from `useQuerySource` (`useIntl` can go here safely).